src/oom/oomd-util.c

```diff
@@ -280,13 +306,35 @@ int oomd_kill_by_swap_usage(Hashmap *h, uint64_t threshold_usage, bool dry_run,
                         continue; /* Try to find something else to kill */
                 }
 
+                dump_until = MAX(dump_until, i);
                 char *selected = strdup(sorted[i]->path);
                 if (!selected)
                         return -ENOMEM;
                 *ret_selected = selected;
-                return r;
+                ret = r;
+                break;
         }
 
+        /* Try dumping top offendors, ignoring any errors that might happen. */
+        _cleanup_free_ char *dump = NULL;
+        _cleanup_fclose_ FILE *f = NULL;
+        size_t size;
+        f = open_memstream_unlocked(&dump, &size);
+        if (!f)
+                return ret;
+
+        fprintf(f, "Considered %d cgroups for killing, top candidates were:\n", n);
+        for (int i = 0; i < dump_until; i++)
+                oomd_dump_swap_cgroup_context(sorted[i], f, "\t");
+
+        r = fflush_and_check(f);
+        if (r < 0)
+                return ret;
+
+        f = safe_fclose(f);
+
+        log_dump(LOG_INFO, dump);
```

Comment:
this appears to be the exact same code as the code above, just the function called in the loop is different? Given theset two functions have the same signature, maybe split out this code into a common helper call that accepts the function to call as parameter?

Reply:
Sounds good to me!